### DISCUSSION TITLE
how to define a route for query string

### DISCUSSION THREAD

**Discussion by WreewanMorhee** [remix-run/remix#7762] [Q&A] [answered]
I have a canonical url like below:
mydomain.com/l?page={number}
number is dynamic from 1 to maybe 100 

how can I define a route for query string ?
and let loader shouldRevalidate based on query string ( not based on pathname )

**Comment by kiliman** [comment]
The routes don't match by query string; they only match against URL path segments.

So in your case, you would create `routes/l.tsx`, and your loader will return the correct data for that page.
```ts
// routes/l.tsx
export async function loader({ request }: DataFunctionArgs) {
  const url = new URL(request)
  const page = url.searchParams.get('page')
  const data = await getData(page)
  return json({ data })
}
```
As for `shouldRevalidate`, Remix will automatically revalidate when search params (query string) changes. Are you saying you don't want that? If so, you have access to the current and next URL to get the search params from.

**Reply by WreewanMorhee** [reply]
how cat I get query string value in `shouldRevalidate` function ??

**Reply by sergiodxa** [reply]
You receive `currentURL` and `nextURL`, from there you can get the search params.

**Reply by WreewanMorhee** [reply]
<img width="594" alt="Screen Shot 2023-10-26 at 2 52 59 AM" src="https://github.com/remix-run/remix/assets/33398071/6d37dfff-f110-4f7a-b8cd-a6e6d180086d">







sorry but ...
You mean this??
I only get `undefined` in both `currentURL` and `nextURL`
and only pathname in both `currentParams` and `nextParams`

( all my remix version are 2.0.1 )

**Comment by Erij-Maherzia-BEN-BRAHIM** [comment, marked as answer]
```js
export const shouldRevalidate: ShouldRevalidateFunction = (options) => {
  if (
    options.formMethod?.toUpperCase() === "GET" &&
    options.currentUrl.pathname === options.nextUrl.pathname &&
    options.currentUrl.search !== options.nextUrl.search
  ) {
    return true ; // return false if you don't want to revalidate when the query change, return true if you want to revalidate when the query change
  }

  return options.defaultShouldRevalidate;
};
```

**Reply by WreewanMorhee** [reply]
I think I typo in my code so cannot get the right one
thank u
